Design.md

```diff
@@ -0,0 +1,41 @@
+# Design Document
+
+This document captures the design decisions made after dicussion in issues. When a design issue is closed, the
+conclusion should be summarized here with a link to the issue.
+
+## Disk Layout
+
+Originally discussed in issue #18
```

Comment:
would be nice if this was a hyperlink to #18 - when I view it rendered it doesn't seem to auto hyperlink

Reply:
sure, I thought it would autolink, but I guess not?